lib/Doctrine/DBAL/Tools/Console/Command/PingCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Doctrine\DBAL\Tools\Console\Command;
+
+use Doctrine\DBAL\Connection;
+use LogicException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+use function is_numeric;
+use function sleep;
+use function sprintf;
+
+class PingCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $defaultName = 'dbal:ping';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Check db is available')
+            ->addOption('limit', null, InputOption::VALUE_REQUIRED, 'Max number of pings to try', '1')
+            ->addOption('sleep', null, InputOption::VALUE_REQUIRED, 'Length of time (seconds) to sleep between pings', '1');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output) : int
+    {
+        $limit = $input->getOption('limit');
+        if (! is_numeric($limit)) {
+            throw new LogicException("Option 'limit' must contain an integer value");
```

Comment:
I'd use `UnexpectedValueException` here, because there is no issue with the code, so it must be a `RuntimeException`

Reply:
I followed the style of https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Tools/Console/Command/RunSqlCommand.php#L61 - I can change it if you want but seems more consistent not to.

Reply:
I think that line might be wrong too, sadly.

https://www.php.net/manual/en/class.runtimeexception.php
https://www.php.net/manual/en/class.logicexception.php

If you read the description for the 2 exception parent classes above what do you conclude?

Reply:
You are probabably right, although its a bit subjective, for example the command can be trigger like it is in testing, with code

```
$this->commandTester->execute([
    '--limit' => '5',
    '--sleep' => '0',
]);
```

I would be inclined to follow the given style, and fix it in a future PR across all the commands.

Reply:
> I would be inclined to follow the given style, and fix it in a future PR across all the commands.
I'm ok with it if you commit to fixing it in a future PR. If not, please don't add more lines to fix.

Reply:
Ok, I've switched this out for `RuntimeException`